src/CMakeLists.txt

```diff
@@ -277,34 +267,35 @@ set_property(TARGET "${PROJECT_NAME}" PROPERTY POSITION_INDEPENDENT_CODE ON)
 # Add the targets we depend on (this does link and include)
 # This propagates to any project that uses this as a dependency
 if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
-    target_link_libraries("${PROJECT_NAME}" "legacy_stdio_definitions") # This won't be required if we rebuild freetype with newer than vs2013
+    target_link_libraries("${PROJECT_NAME}" PRIVATE "legacy_stdio_definitions") # This won't be required if we rebuild freetype with newer than vs2013
 endif()
 
 # If gcc < 9, needs to link against stdc++fs
 if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.0)
-    target_link_libraries("${PROJECT_NAME}" stdc++fs)
+    target_link_libraries("${PROJECT_NAME}" PRIVATE stdc++fs)
 endif()
 
-# Strip trailiing whitespace from SDL2_Libraries for Ubuntu1604 / libsdl2-dev 2.0.4
-string(STRIP "${SDL2_LIBRARIES}" SDL2_LIBRARIES)
-target_link_libraries("${PROJECT_NAME}" "${SDL2_LIBRARIES}")
-# Link against GLEW, via GLEW::GLEW which is an alias for GLEW::glew or GLEW::glew_s based on availability. 
-if(TARGET GLEW::GLEW)
-    target_link_libraries("${PROJECT_NAME}" GLEW::GLEW)
-else()
-    # Fallback to the variable for our partial windows CMake config file. We should add an imported target alias to this.
-    target_link_libraries("${PROJECT_NAME}" "${GLEW_LIBRARIES}")
+target_link_libraries("${PROJECT_NAME}" PRIVATE glm::glm)
+# GLEW::GLEW which is an alias for GLEW::glew or GLEW::glew_s based on availability.
+target_link_libraries("${PROJECT_NAME}" PRIVATE GLEW::GLEW)
+target_link_libraries("${PROJECT_NAME}" PRIVATE SDL2::SDL2)
+target_link_libraries("${PROJECT_NAME}" PRIVATE "${IL_LIBRARIES}")
+target_link_libraries("${PROJECT_NAME}" PRIVATE OpenGL::GL)
+target_link_libraries("${PROJECT_NAME}" PRIVATE Threads::Threads)
+# imgui must be public as the  headers are visible in flamegpu/FLAMEGPU2
```

Comment:
Are they visible out of necessity, or me being crappy?

Reply:
https://github.com/FLAMEGPU/FLAMEGPU2-visualiser/blob/3c4d209a0fa8008fbd398095a66afcf085d1bf61/include/flamegpu/visualiser/config/ImGuiWidgets.h#L24

It might be possible to move these to a `.cpp` as it's a forced template instantiation?

Reply:
There's also various calls to ImGui which could be moved to cpp, they were left in header because lazy (or inline).

Reply:
Not sure if it's worth addressing these (or making an issue). Do we care?

Reply:
https://github.com/FLAMEGPU/FLAMEGPU2-visualiser/issues/111